Parameters are not evaluated properly for Qualitygates.Search

The appropriate parameter for Quality Gate search requests is:

`gateId`

![image](https://user-images.githubusercontent.com/32190247/182005150-c9e829c2-83ef-4067-882d-2b192f3a9880.png)

However the following code snippet creates the parameter `gate_id`:
https://github.com/reinoudk/go-sonarcloud/blob/85ad2437bd4dfff787e6a8f10481f53c43053840/sonarcloud/sonarcloud.go#L173

I have verified this by finding this request object in the debugger while running the `Create` function in `resource_quality_gate_selection_test.go`
```
Method: "GET",
        URL: *net/url.URL {
                Scheme: "https",
                Opaque: "",
                User: *net/url.Userinfo nil,
                Host: "sonarcloud.io",
                Path: "/api/qualitygates/search",
                RawPath: "",
                ForceQuery: false,
                RawQuery: "=&=&=&=&gate_id=66749&organization=arnav-bhutani&organization=arnav-bhutani",
                Fragment: "",
                RawFragment: "",},
        Proto: "HTTP/1.1",
...
```
The error message I encountered is as follows:
```
=== RUN   TestAccResourceQualityGateSelection
    .../terraform-provider-sonarcloud/sonarcloud/resource_quality_gate_selection_test.go:25: Step 1/1 error: Error running apply: exit status 1

        Error: Could not read Quality Gate Selection

          with sonarcloud_quality_gate_selection.test,
          on terraform_plugin_test.tf line 2, in resource "sonarcloud_quality_gate_selection" "test":
           2: resource "sonarcloud_quality_gate_selection" "test" {

        The Search request reutrned an error: received non 2xx status code (400): The
        'gateId' parameter is missing
--- FAIL: TestAccResourceQualityGateSelection (4.51s)
FAIL
FAIL    terraform-provider-sonarcloud/sonarcloud        4.873s


> Test run finished at 7/30/2022, 5:25:27 PM <
```


As a side note, it looks like the organization parameter is duplicated; however this doesn't appear to be a blocking issue.

This issue blocks https://github.com/rewe-digital/terraform-provider-sonarcloud/issues/76 

**EDITS**: Added error message

--- a/sonarcloud/sonarcloud.go
+++ b/sonarcloud/sonarcloud.go
@@ -8,11 +8,12 @@ package sonarcloud
 
 import (
 	"fmt"
-	"github.com/iancoleman/strcase"
 	"io"
 	"net/http"
 	"reflect"
 	"strconv"
+
+	"github.com/iancoleman/strcase"
 )
 
 // API FIXME: this should be injected
@@ -170,7 +171,7 @@ func paramsFrom(items ...interface{}) []string {
 				fieldValue = strconv.FormatBool(v.Field(i).Interface().(bool))
 			}
 
-			params[j] = strcase.ToSnake(t.Field(i).Name)
+			params[j] = strcase.ToCamel(t.Field(i).Name)
 			params[k] = fieldValue
 		}
 

--- a/sonarcloud/sonarcloud.go
+++ b/sonarcloud/sonarcloud.go
@@ -171,7 +171,7 @@ func paramsFrom(items ...interface{}) []string {
 				fieldValue = strconv.FormatBool(v.Field(i).Interface().(bool))
 			}
 
-			params[j] = strcase.ToCamel(t.Field(i).Name)
+			params[j] = strcase.ToLowerCamel(t.Field(i).Name)
 			params[k] = fieldValue
 		}
 

